linux-tools/test_time/test_time.py

```diff
@@ -0,0 +1,195 @@
+        if utils.re.search(
+                "(real\s+%s.\d+)\nuser\s+\d+.\d+\nsys\s+\d.\d+" %
+                self.seconds,
+                time_output):
+            utils.logging.info("Verified '--portabaility' option successfuly.")
+        else:
+            self.nfail += 1
+            raise error.TestError(
+                "\n Failed To verify --portability option of time utility")
+
+    def test_exit_status_of_cmd(self, cmd, expected_status):
+        """
+        Method Name:test_exit_status_of_cmd
+        This method will verify the exit status of the command invoked using time utility
+        Ex:/usr/bin/time -f "%x is Exit status od the cmd %C" <cmd>
+        """
+        cmd_to_execute = "%s " % self.time_exe + \
+            " -f '%x is the Exit status of the cmd %C' " + cmd
+        search_pattren = "%s is the Exit status of the cmd %s" % (
+            expected_status, cmd)
+        time_output = utils.commands.getoutput(cmd_to_execute)
+        if utils.re.search("%s" % search_pattren, time_output):
+            utils.logging.info(
+                "Verified Exist status using time command successfuly.")
+        else:
+            self.nfail += 1
+            raise error.TestError(
+                "\n Failed To verify --portability option of time utility")
+
+    def test_output_option(self):
+        """
+        Method Name:test_output_file
+        This method will verify the whether output file is created or not using '-o' option
+        Ex:/usr/bin/time -o time_out.txt <cmd>
+        """
+        cmd_invoked = "sleep %s" % self.seconds
+        self.output_file = '%s/time-output-%s' % (
+            self.tmpdir, utils.time.strftime('%Y-%m-%d-%H.%M.%S'))
+        cmd = "%s -v -o %s %s" % (self.time_exe, self.output_file, cmd_invoked)
+        if (utils.system(cmd) != 0):
+            self.nfail += 1
+            raise error.TestError(
+                '\n Time command with -o option is failed to execute')
+        else:
+            if utils.os.path.exists(
+                    self.output_file) and utils.file_contains_pattern(
+                    self.output_file, cmd_invoked):
+                utils.logging.info(
+                    "Output file is created successfully using -o option of time utility")
+            else:
+                self.nfail += 1
+                raise error.TestError(
+                    "\n Failed To create output file using -o option")
+
+    def test_append_option(self):
+        """
+        Method Name: verify_append_option
+        This Method will verify the --append option of time utility
+        Ex:/usr/bin/time -v -a -o time_out.txt <cmd>
+        """
+        cmd_invoked = "pwd"
+        cmd = "%s -v -a -o %s %s" % (self.time_exe,
+                                     self.output_file, cmd_invoked)
+        if (utils.system(cmd) != 0):
+            self.nfail += 1
+            raise error.TestError(
+                '\n time command with -a option is failed to execute')
```

Comment:
As you may want to check command exit status every time, why not calling ``test_exit_status_of_cmd```(maybe need a bit of update)